aya/src/programs/mod.rs

```diff
@@ -397,6 +397,14 @@ fn load_program(prog_type: bpf_prog_type, data: &mut ProgramData) -> Result<(),
         ..
     } = obj;
 
+    let target_kernel_version = match *kernel_version {
+        KernelVersion::Any => {
+            let (major, minor, patch) = crate::sys::kernel_version().unwrap();
+            KernelVersion::Version((major << 16) + (minor << 8) + patch)
+        }
+        _ => *kernel_version,
+    };
```

Comment:
I don't think we need to wrap here. We can return `(major << 16) + (minor << 8) + patch` in the `Any` branch, and `v.into()` in the `Version(v)` branch